src/test/java/seedu/address/model/RoomBookTest.java

```diff
@@ -0,0 +1,94 @@
+package seedu.address.model;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.PriorityQueue;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.model.hotel.Room;
+
+
+
+class RoomBookTest {
+
+    @Test
+    void testEquals() {
+        PriorityQueue<Room> rooms = new PriorityQueue<>();
+        Room[] arrayOfRooms = new Room[10];
+        RoomBook roomBook1 = new RoomBook(rooms, arrayOfRooms, 10);
+        RoomBook roomBook2 = new RoomBook(rooms, arrayOfRooms, 10);
+        //same object -> returns true
+        assertTrue(roomBook1.equals(roomBook1));
+
+        //diff object, but same fields -> returns true
+        assertTrue(roomBook1.equals(roomBook2));
+
+        //null object -> returns false
+        assertFalse(roomBook1.equals(null));
+
+        roomBook2 = new RoomBook(rooms, arrayOfRooms, 10);
+        PriorityQueue<Room> rooms1 = new PriorityQueue<>();
+        rooms1.add(new Room(1));
+        roomBook2.setRooms(rooms1);
+        //different Priority Queue of rooms -> returns false
+        assertFalse(roomBook1.equals(roomBook2));
+
+        roomBook2 = new RoomBook(rooms, arrayOfRooms, 10);
+        roomBook2.setNumOfRooms(100);
+        //different number of rooms -> returns false
+        assertFalse(roomBook1.equals(roomBook2));
+
+        roomBook2 = new RoomBook(rooms, arrayOfRooms, 10);
+        roomBook2.setRoomsInArray(new Room[100]);
+        //different array of rooms -> returns false
+        assertFalse(roomBook1.equals(roomBook2));
+    }
+
+    @Test
```

Comment:
hmm, i dont really think there is a need to test if the priority queues are equal? or if you want to test, you can just
``int[] array1 = roombook1.toArray();``
``int[] array2 = roombook2.toArray()``
``Arrays.equal(array1, array2)``

Reply:
Okay